homework01/allan/vector.h

```diff
@@ -0,0 +1,43 @@
+class Vector
+{
+public:
+
+	Vector()
+	:Vector(0,0)
+	{
+	}
+
+	Vector(int x, int y)
+	:m_x(x),
+	 m_y(y)
+	{
+	}
+
+	bool operator==(const Vector& rhs) const
+	{
+		return m_x == rhs.m_x &&
+			   m_y == rhs.m_y;
+	}
+
+	int GetX() const
+	{
+		return m_x;
+	}
+
+	int GetY() const
+	{
+		return m_y;
+	}
+
+	Vector& Add(const Vector& rhs)
+	{
+		m_x += rhs.m_x;
+		m_y += rhs.m_y;
+		return *this;
```

Comment:
Same comment as above: this should go to the cc file. The exception is inlined methods (for performance): you need to keep them in the header file, but if you need those, it's better to define them outside the class declaration.

Reply:
Thanks David. I've got 2 cc files now. In this case I either must put definitions to the cc file or else inline right? Else I'll get a duplicate at link time?